### PR TITLE
ADEN-10401 Templates 2.0 Improvements

### PR DESCRIPTION
**Pull request by Bielik20**

### CODE REVIEW

File: platforms/shared/templates/handlers/bfaa/bfaa-bootstrap-handler.ts [hunk 1]
```diff
@@ -52,4 +52,9 @@ export class BfaaBootstrapHandler implements TemplateStateHandler {
 		this.adSlot.show();
 		document.body.classList.add('has-uap');
 	}
+
+	async onDestroy(): Promise<void> {
+		document.body.classList.remove('has-uap');
+		this.adSlot.getAdContainer().classList.remove('iframe-container'); // TODO: is it necessary?
```
Comment (fraszczakszymon) [inline platforms/shared/templates/handlers/bfaa/bfaa-bootstrap-handler.ts:58]: It is not. Whole container is going to be removed on slot destroy.

File: platforms/shared/templates/handlers/bfab/bfab-bootstrap-handler.ts [hunk 2]
```diff
@@ -43,4 +43,8 @@ export class BfabBootstrapHandler implements TemplateStateHandler {
 	async onLeave(): Promise<void> {
 		this.adSlot.show();
 	}
+
+	async onDestroy(): Promise<void> {
+		this.adSlot.getAdContainer().classList.remove('iframe-container'); // TODO: is it necessary?
```
Comment (fraszczakszymon) [inline platforms/shared/templates/handlers/bfab/bfab-bootstrap-handler.ts:48]: Ditto

File: platforms/shared/templates/handlers/close-to-hidden-button-handler.ts [hunk 3]
```diff
@@ -19,6 +19,7 @@ export class CloseToHiddenButtonHandler implements TemplateStateHandler {
 	constructor(@Inject(TEMPLATE.SLOT) private adSlot: AdSlot, private helper: CloseButtonHelper) {}
 
 	async onEnter(transition: TemplateTransition<'hidden'>): Promise<void> {
+		// TODO: There is no logic that removes that onClick callback, will it be deleted alongside button element?
```
Comment (fraszczakszymon) [inline platforms/shared/templates/handlers/close-to-hidden-button-handler.ts:22]: I believe so. In my opinion, we don't need to handle that

File: platforms/shared/templates/handlers/roadblock/roadblock-handler.ts [hunk 4]
```diff
@@ -29,11 +29,10 @@ export class RoadblockHandler implements TemplateStateHandler {
 		@Inject(ROADBLOCK_CONFIG) private config: RoadblockConfig,
 	) {}
 
+	// TODO: like BFAA
```
Comment (fraszczakszymon) [inline platforms/shared/templates/handlers/roadblock/roadblock-handler.ts:32]: What about this todo?

File: platforms/shared/templates/handlers/video/video-bootstrap-handler.ts [hunk 5]
```diff
@@ -32,34 +34,40 @@ export class VideoBootstrapHandler implements TemplateStateHandler {
 			return this.playerRegistry.discard();
 		}
 
-		this.adSlot.addClass('theme-hivi'); // Required by replay-overlay
-		this.playerRegistry.register();
+		this.adSlot.addClass('theme-hivi'); // Required by replay-overlay TODO: shouldn't it be in bfaa-bootstrap-handler
```
Comment (fraszczakszymon) [inline platforms/shared/templates/handlers/video/video-bootstrap-handler.ts:37]: Agree, we can move it.

File: platforms/shared/templates/handlers/video/video-bootstrap-handler.ts [hunk 6]
```diff
@@ -32,34 +34,40 @@ export class VideoBootstrapHandler implements TemplateStateHandler {
...
 					(event: { status: string }) => event.status === universalAdPackage.SLOT_FORCE_UNSTICK,
 				),
-			)
-			.subscribe(() => player.stop());
+				tap(() => player.stop()),
+			),
+		);
 	}
 
+	// TODO: should we clean after that as well?
```
Comment (fraszczakszymon) [inline platforms/shared/templates/handlers/video/video-bootstrap-handler.ts:70]: Good point. Should we create a separate ticket for that?

File: platforms/ucp/templates/handlers/logo-replacement/logo-replacement-ucp-handler.ts [hunk 7]
```diff
@@ -9,6 +9,7 @@ export class LogoReplacementUcpHandler implements TemplateStateHandler {
 		@Inject(TEMPLATE.PARAMS) private params: LogoReplacementParams,
 	) {}
 
+	// TODO: What about this? (+ other platforms)
```
Comment (fraszczakszymon) [inline platforms/ucp/templates/handlers/logo-replacement/logo-replacement-ucp-handler.ts:12]: I don't think it's crucial for non-SPA platforms

File: platforms/ucp/templates/handlers/bfaa/bfaa-oasis-config-handler.ts [hunk 8]
```diff
@@ -11,6 +11,7 @@ import { Inject, Injectable } from '@wikia/dependency-injection';
 export class BfaaOasisConfigHandler implements TemplateStateHandler {
 	constructor(@Inject(TEMPLATE.PARAMS) private params: UapParams) {}
 
+	// TODO: Should we clean after that somehow (+ other platforms)?
```
Comment (fraszczakszymon) [inline platforms/ucp/templates/handlers/bfaa/bfaa-oasis-config-handler.ts:14]: It's worth to dig into universalAdPackage and check what's there, but I don't think it's crucial for non-SPA platforms

File: platforms/ucp/templates/handlers/logo-replacement/logo-replacement-ucp-handler.ts (hunk 7 above)
Comment (Bielik20) [reply platforms/ucp/templates/handlers/logo-replacement/logo-replacement-ucp-handler.ts:12]: Yeah, but what about mobile wiki? I don't remember anything that would compensate for that there.

File: platforms/ucp/templates/handlers/bfaa/bfaa-oasis-config-handler.ts (hunk 8 above)
Comment (Bielik20) [reply platforms/ucp/templates/handlers/bfaa/bfaa-oasis-config-handler.ts:14]: I don't think any of this is crucial, or even necessary, for non-SPA platforms. What I try to do here is to make them handlers (in general) capable of being destroyed, and thus used on SPA platforms.

File: platforms/shared/templates/handlers/roadblock/roadblock-handler.ts (hunk 4 above)
Comment (Bielik20) [reply platforms/shared/templates/handlers/roadblock/roadblock-handler.ts:32]: https://wikia-inc.atlassian.net/browse/ADEN-10430

File: platforms/ucp/templates/handlers/logo-replacement/logo-replacement-ucp-handler.ts (hunk 7 above)
Comment (Bielik20) [reply platforms/ucp/templates/handlers/logo-replacement/logo-replacement-ucp-handler.ts:12]: https://wikia-inc.atlassian.net/browse/ADEN-10430

File: platforms/ucp/templates/handlers/bfaa/bfaa-oasis-config-handler.ts (hunk 8 above)
Comment (Bielik20) [reply platforms/ucp/templates/handlers/bfaa/bfaa-oasis-config-handler.ts:14]: https://wikia-inc.atlassian.net/browse/ADEN-10430

File: platforms/shared/templates/handlers/video/video-bootstrap-handler.ts (hunk 6 above)
Comment (Bielik20) [reply platforms/shared/templates/handlers/video/video-bootstrap-handler.ts:68]: https://wikia-inc.atlassian.net/browse/ADEN-10431

**Pull request merged**